### PR TITLE
refactor: Replaces all hardcoded strings with @string resource (#412) 

### PR DESCRIPTION
**Pull request by coderaashir**
This  is done only for the Activity and Fragment files and all XML files.

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 1]
```diff
@@ -239,7 +239,7 @@ public void onNothingSelected(AdapterView<?> parent) {
 
     @Override
     public void showChargesCreatedSuccessfully(Charges changes) {
-        Toast.makeText(getActivity(), "Charge created successfully", Toast.LENGTH_LONG)
+        Toast.makeText(getActivity(), R.string.change_created_success, Toast.LENGTH_LONG)
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: change to ```charge_created_successfully```

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/loanchargedialog/LoanChargeDialogFragment.java [hunk 2]
```diff
@@ -247,7 +247,7 @@ public void onNothingSelected(AdapterView<?> parent) {
 
     @Override
     public void showLoanChargesCreatedSuccessfully(Charges charges) {
-        Toast.makeText(getActivity(), "Charge created successfully", Toast
+        Toast.makeText(getActivity(), R.string.charge_created, Toast
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/loanchargedialog/LoanChargeDialogFragment.java]: Use above string resources "charge_created_successfully"

Comment (coderaashir) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/loanchargedialog/LoanChargeDialogFragment.java]: I changed it to "charge_created", I hope thats fine

Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/loanchargedialog/LoanChargeDialogFragment.java]: Still, there are two string resources with same value 
1. ```<string name="charge_created_success">Charge created successfully</string>```
2. ```<string name="charge_created">Charge created successfully</string>```

Stay with 1st string and use this.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/clientcharge/ClientChargeFragment.java [hunk 3]
```diff
@@ -213,7 +213,7 @@ public void showChargesList(Page<Charges> chargesPage) {
             // Clients Available for fetch
             if (chargesPage.getPageItems().size() == 0 &&
                     (chargesPage.getTotalFilteredRecords() == chargesList.size()))
-                Toaster.show(rootView, "No more Charges Available");
+                Toaster.show(rootView, getString(R.string.no_more_changes_available));
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/clientcharge/ClientChargeFragment.java]: String name should be ```no_more_charges_available``` not  ```no_more_changes_available```

File: mifosng-android/src/main/res/layout/row_collection_list_group_client.xml [hunk 4]
```diff
@@ -18,7 +18,7 @@
         <TextView
             android:layout_width="0dp"
             android:layout_height="wrap_content"
-            android:text="237"
+            android:text="@string/_237"
```
Comment (coderaashir) [inline mifosng-android/src/main/res/layout/row_collection_list_group_client.xml]: client_name has been defined as "Client Name", not "Raj Kumar Kaushikl"

**Comment by therajanmaurya** [comment]
PR looks Good there is one things need to change.
1. change commit message to "refactor: your message". you can use git command 
```git commit --amend```

@ishan1604 : @coderaashir have committed the main master branch two commits in this PR.
What you think, he has to rebase his branch or this can be fine for now.

**Comment by coderaashir** [comment]
@therajanmaurya commit message has been changed
what do you mean by 2 commits? I can only see one, i rebased and squashed all the additional commits some time back

**Comment by therajanmaurya** [comment]
I mean commit message, not PR message, and check your changed files. you have added the two more commits of main master branch, In file DashboardActivity.java and CenterListFragment.java.

**Comment by coderaashir** [comment]
@therajanmaurya I've changed both messages 
Should I remove that commit? I updated my branch after I had made the PR, i think

**Comment by therajanmaurya** [comment]
Still commit message is ```Puts all hardcore style strings in resources folder... ```
If you will change the messge then it will look like this
```refactor: Puts all hardcore style strings in resources folder... ```

**Comment by coderaashir** [comment]
Sorry, I was running the command from another branch @therajanmaurya

File: mifosng-android/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -137,6 +137,40 @@
...
+    <string name="approve_loan">Approve Loan</string>
+    <string name="disburse_loan">Disburse Loan</string>
+    <string name="loan_closed">Loan Closed</string>
+    <string name="working">Working...</string>
+    <string name="updating_gps">Updating Gps Data</string>
+    <string name="save_pin">Save pin</string>
+    <string name="survey_name">Survey Name</string>
+    <string name="start_tracking">Start tracking</string>
+    <string name="stop_tracking">Stop tracking</string>
+    <string name="client_id">237</string>
```
Comment (ishan1604) [inline mifosng-android/src/main/res/values/strings.xml]: I don't quite like this string here. 

If this is just for debugging purposes.

You should use ```tools:text``` attribute and you can hardcode your string there

File: mifosng-android/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -137,6 +137,40 @@
...
+    <string name="_12_02_2014">12-02-2014</string>
```
Comment (ishan1604) [inline mifosng-android/src/main/res/values/strings.xml]: Same problem with this string

**Comment by coderaashir** [comment]
@therajanmaurya regarding the conflics: should I resolve them myself or should I consult with you?

**Comment by coderaashir** [comment]
Like in MifosBaseFragment, for some I'm getting a build error in line 73: 

SavingsAccountSummaryFragment savingsAccountSummaryFragment
                = SavingsAccountSummaryFragment.newInstance(419, depositType);

(you can find more info in gitter)
 
So what should I pass as the boolean in the arguments?

**Comment by therajanmaurya** [comment]
@ishan1604 Please the final look and merge

**Comment by luckyman20** [comment]
@coderaashir Please squash all the commits and solve the conflicts

**Comment by droidchef** [comment]
The PR cannot be merged due to conflicts. They must be resolved before we can review the PR again and eventually merge it.

**Pull request closed without merging**